File: frontend/src/interface/variant.ts
import type { ColorInterface } from "./color";
import type { SizeInterface} from "./size";
import type { ProductInterface} from "./product"

export interface VariantInterface {
    id?: number
    ProductID?:	number    
    Product?:   ProductInterface
    color?:	    ColorInterface
    size?:      SizeInterface
    quantity?:  number
    picture?:   string    
}

File: backend/entity/variant.go
package entity

import (
   "gorm.io/gorm"
)

type Variant struct {
   gorm.Model
   ProductID 	uint    	`json:"product_id"`
   Product     *Product    `gorm:"foreignKey:ProductID" json:"product"`
   ColorID	   uint     	`json:"color_id"`
   Color		   *Color	   `gorm:"foreignKey:ColorID" json:"color"`
   SizeID      uint     	`json:"size_id"`
   Size		   *Size	      `gorm:"foreignKey:SizeID" json:"size"`
   Quantity    uint       `json:"quantity"`
   Picture     string      `json:"picture"`
}